tests/api/src/test/java/APITesting_WCOrder.java

```diff
@@ -0,0 +1,314 @@
+            queryParam("_method", method).
+            body(jsonObj.toString()).
+        when().
+            post().
+        then().
+            statusCode(200);
+
+        // Get Status.
+        given().
+            spec(this.mRequestSpec).
+            header("Content-Type", ContentType.JSON).
+            queryParam("path", path).
+            body(jsonObj.toString()).
+        when().
+            get().
+        then().
+            statusCode(200).
+            body("data.status", equalTo("processing"));
+
+        // Set status to completed
+        jsonBody.put("status", "completed");
+        jsonObj.put("body", jsonBody.toString());
+        given().
+            spec(this.mRequestSpec).
+            header("Content-Type", ContentType.JSON).
+            queryParam("path", path).
+            queryParam("_method", method).
+            body(jsonObj.toString()).
+        when().
+            post().
+        then().
+            statusCode(200).
+            body("data.status", equalTo("completed"));
+    }
+
+    @Test
+    public void canGetOrderNotes() {
+        given().
+            spec(this.mRequestSpec).
+            queryParam("path", "/wc/v4/orders/591/notes").
+        when().
+            get().
+        then().
+            statusCode(200).
+            body("data", hasSize(2),
+                "data.find { it.id == 1173 }.note", equalTo("Order status changed from Processing to Completed.")
+            );
+    }
+
+    @Test
+    public void canAddOrderNote() {
+        String path = "/wc/v4/orders/565/notes";
+        String method = "post";
+
+        JSONObject jsonBody = new JSONObject();
+        jsonBody.put("note", "Adding note to an order test");
+        jsonBody.put("customer_note", false);
+        jsonBody.put("added_by_user", true);
+        JSONObject jsonObj = new JSONObject();
+        jsonObj.put("path", path);
+        jsonObj.put("body", jsonBody.toString());
+        jsonObj.put("json", "true");
+        jsonObj.put("method", method);
+
+        given().
+            spec(this.mRequestSpec).
+            header("Content-Type", ContentType.JSON).
+            queryParam("path", path).
+            queryParam("_method", method).
+            body(jsonObj.toString()).
+        when().
+            post().
+        then().
+            statusCode(200);
+    }
+
+    @Test
+    public void canGetOrderShipmentTrackings() {
+        given().
+            spec(this.mRequestSpec).
+            queryParam("path", "/wc/v4/orders/635/shipment-trackings&_fields=" + mTrackingFields).
+        when().
+            get().
+        then().
+            statusCode(200).
+            body("data", hasSize(1),
+                "data[0].tracking_number", equalTo("324567890234567")
+            );
+    }
+
+    @Test
+    public void canGetOrderShipmentProviders() {
+        given().
+            spec(this.mRequestSpec).
+            queryParam("path", "/wc/v4/orders/635/shipment-trackings/providers").
```

Comment:
As with the `orders/<id>/shipment-trackings` endpoint, it looks like `orders/<id>/shipment-trackings/providers` also isn't supported in `v4`. It would be good to double check the expected behavior but for now using `v3` resolves the failure I'm seeing:

```suggestion
            queryParam("path", "/wc/v3/orders/635/shipment-trackings/providers").
```